pkg/discovery/util/pod_util.go

```diff
@@ -137,3 +137,29 @@ func GroupPodsByNode(pods []*api.Pod) map[string][]*api.Pod {
 	}
 	return podsNodeMap
 }
+
+// GetReadyPods returns pods that in Ready status
+func GetReadyPods(pods []*api.Pod) []*api.Pod {
+	readyPods := []*api.Pod{}
+
+	for _, pod := range pods {
+		if podIsReady(pod) {
+			readyPods = append(readyPods, pod)
+		} else {
+			glog.V(4).Infof("Pod %s is not ready", pod.Name)
+		}
+	}
+
+	return readyPods
+}
+
+// PodIsReady checks if a pod is in Ready status.
+func podIsReady(pod *api.Pod) bool {
```

Comment:
Should make sure  `condition.Type == api.PodReady` for all the conditions.
For example, for pod: `default/router-1-25zn2`, which is  in `Unknown` state:
<img width="795" alt="screen shot 2018-03-12 at 4 24 39 pm" src="https://user-images.githubusercontent.com/27221807/37307657-08bfea9e-2612-11e8-9803-ddcf01e187b5.png">

I think we may check the `Pod.Status.ContainerStatuses.Ready` for all the containers in the Pod.

Reply:
<img width="777" alt="screen shot 2018-03-12 at 4 22 30 pm" src="https://user-images.githubusercontent.com/27221807/37307827-867f5c76-2612-11e8-8745-8db76ad946c6.png">

another pod in `Error` state, and its container is not running.

Reply:
Talked with @maxwangvmt . I misunderstood it. This check works 👍